mcs/tools/mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -5662,4 +5666,85 @@ protected override StringBuilder AppendNamespace (StringBuilder buf, TypeReferen
 	}
 }
 
+internal abstract class FlagsFormatter {
+	public abstract string Format(long c, Dictionary<long, string> values, string typename);
```

Comment:
What I think might make more sense is to:

1. Make `FlagsFormatter` a "normal" (non-`abstract`) class with a `virtual` `Format()` method.
2. `Format()` is `public virtual string Format(object value, TypeReference valueType)`

Then, _most_ of `MakeAttributesValueString()` becomes the base method implementation, and then it can be specialized:

    partial class FlagsFormatter {
        public string Format (object value, TypeReference valueType)
        {
            // existing MakeAttributesValueString() body, lines 2350-2365
            return Format (value, valueType.Resolve ());
        }

        protected virtual string Format (object value, TypeDefinition valueDef)
        {
            // existing MakeAttributesValueString() body lines 2365-end 
            // _except_ the enum-handling code on lines 2368 ... `return "(" ...`.
        }
    }

    class StandardFlagsEnumFormatter : FlagsFormatter {
        protected override string Format (object value, TypeDefinition valueDef)
        {
            if (valueDef.CustomAttributes.Any (ca => ca.AttributeType.FullName == "System.FlagsAttribute")) {
                return string.Join (" | ",
                        (from i in values.Keys
                         where (c & i) != 0
                         select typename + "." + values [i])
                        .DefaultIfEmpty (v.ToString ()).ToArray ());
            }
            return base.Format (value, valueDef);
    }